### ISSUE TITLE
Support branch settings in goodtables.yml

### ISSUE THREAD

**Issue by roll** [frictionlessdata/goodtables.io#81]
# Overview

Now github plugin could work with PRs and different branches so branch filter could be added to `goodtables.yml` (see Travis CI approach)

**Comment by pwalsh** [comment]
Do we currently hardcode to "master", or, do we currently use whatever GH has as the `default`?

**Comment by amercader** [comment]
@pwalsh we create jobs whenever we receive a payload from GitHub on the webhook, so right now is all commits in all branches and PRs. Travis approach of a `branches > except` or `branches > only` option in `goodtables.yml` seems the most straightforward. 

With the current implementation the `goodtables.yml` file is parsed as part of the async task chain, once the job is started so if the branch does not match the ones that should be build we would need to cancel job and it would appear in the UI (we would probably need to add a new "Cancelled" status). Not sure how Travis handles this but perhaps my expected behaviour would be to not see these cancelled builds at all, and not store a job for them as it would be just noise so maybe we just cancel and delete the job object.

**Comment by amercader** [comment]
Not a huge priority for Beta for me

**Comment by andylolz** [comment]
I manage datasets using internal PRs… So I’d like the goodtables badge I use in the README to always show the master branch status. I don’t mind a job per payload (it’s good, in fact!) but I’d like to be able to filter the results by branch, like this:

```
https://goodtables.io/badge/github/org/repo.svg?branch=master
```

**Comment by amercader** [comment]
Good idea @andylolz, I've created a new issue (#265) as this is related but not quite the same issue as this
